extensions/cornerstone/src/init.js

```diff
@@ -73,61 +52,40 @@ export default function init({ servicesManager, configuration = {} }) {
   initCornerstoneTools(defaultCsToolsConfig);
 
   // ~~ Toooools 🙌
-  const {
-    PanTool,
-    ZoomTool,
-    WwwcTool,
-    MagnifyTool,
-    StackScrollTool,
-    StackScrollMouseWheelTool,
-    // Touch
-    PanMultiTouchTool,
-    ZoomTouchPinchTool,
-    // Annotations
-    EraserTool,
-    BidirectionalTool,
-    LengthTool,
-    AngleTool,
-    FreehandRoiTool,
-    EllipticalRoiTool,
-    DragProbeTool,
-    RectangleRoiTool,
-    // Segmentation
-    BrushTool,
-  } = csTools;
   const tools = [
-    PanTool,
-    ZoomTool,
-    WwwcTool,
-    MagnifyTool,
-    StackScrollTool,
-    StackScrollMouseWheelTool,
+    csTools.PanTool,
+    csTools.ZoomTool,
+    csTools.WwwcTool,
+    csTools.MagnifyTool,
+    csTools.StackScrollTool,
+    csTools.StackScrollMouseWheelTool,
     // Touch
-    PanMultiTouchTool,
-    ZoomTouchPinchTool,
+    csTools.PanMultiTouchTool,
+    csTools.ZoomTouchPinchTool,
     // Annotations
-    EraserTool,
-    BidirectionalTool,
-    LengthTool,
-    AngleTool,
-    FreehandRoiTool,
-    EllipticalRoiTool,
-    DragProbeTool,
-    RectangleRoiTool,
+    csTools.ArrowAnnotateTool,
+    csTools.EraserTool,
+    csTools.BidirectionalTool,
+    csTools.LengthTool,
+    csTools.AngleTool,
+    csTools.FreehandRoiTool,
+    csTools.EllipticalRoiTool,
+    csTools.DragProbeTool,
+    csTools.RectangleRoiTool,
     // Segmentation
-    BrushTool,
+    csTools.BrushTool,
   ];
 
-  tools.forEach(tool => csTools.addTool(tool));
-
-  csTools.addTool(csTools.ArrowAnnotateTool, {
-    configuration: {
-      getTextCallback: (callback, eventDetails) =>
-        callInputDialog(null, eventDetails, callback),
-      changeTextCallback: (data, eventDetails, callback) =>
-        callInputDialog(data, eventDetails, callback),
-    },
-  });
+  if (!isEmpty(configuration.tools)) {
+    /* Add tools with its custom props through extension configuration. */
+    tools.forEach(tool => {
+      const toolName = tool.name.replace('Tool', '');
+      const props = configuration.tools[toolName] || {};
+      csTools.addTool(tool, props);
+    });
```

Comment:
Are we setting `tools` to `configuration.tools`? I notice that, despite the check, we still hook up our locally defined ones? We may also want this to support the same config / abilities as the `react-cornerstone-viewport` tools prop.

Reply:
configuration.tools is the default configuration, Inside configuration tools object, each key is a different tool and the user can add any prop that will be passed as arg to addTool.